src/dashboard.tsx

```diff
@@ -354,13 +372,13 @@ export class URLInput extends Widget {
     });
   }
 
+  public _url = '';
```

Comment:
Rather than making these public, let's find out a way to get all the info out that we need. The `URLInput` widget already has a public getter for the URL, so this change isn't necessary. We could also make the dashboard responsible for creating the server connection settings so that it is available to it. Those settings can then be passed in as constructor options to the URL input.

Reply:
Can't the `DaskDashboardLauncher` just get its own instance of `ServerConnection.makeSettings()`?

Reply:
I had a feeling this would be the stickiest spot. Reverted to old protection status!

Reply:
It could make its own settings, though in theory I suppose we might want to ensure that the two widgets are using the same server connection settings. I'm unsure when it might be a problem if they were different, however.